shared/src/iosMain/kotlin/com/karumi/gallery/domain/TimeProvider.kt

```diff
@@ -0,0 +1,36 @@
+package com.karumi.gallery.domain
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.ptr
+import platform.Foundation.NSUserDefaults
+import platform.posix.gettimeofday
+import platform.posix.timeval
+
+actual class TimeProvider {
+
+  companion object {
+    private const val TIME_ARG = "time_arg"
+    private const val THREE_MONTHS_IN_MILLIS = 259200
+  }
+
+  private val delegate: NSUserDefaults = NSUserDefaults.standardUserDefaults
```

Comment:
Where does the `delegate` name comes from? In iOS a delegate is just a listener and it doesn't look like we are using that pattern in here. What about just calling it `userDefaults`?

Reply:
I don't know too much in iOS and I guess it was something that it's used in iOS development, I'll change it! 👍